src/quo/components/slideshow/slider_bar/view.cljs

```diff
@@ -0,0 +1,168 @@
+  [{:keys [active-index]}]
+  (cond
+    (= active-index ITEMS_BEFORE_SCROLL) MEDIUM_SCALE
+    (> active-index ITEMS_BEFORE_SCROLL) SMALL_SCALE
+    :else                                DEFAULT_SCALE))
+
+(defn- second-item
+  "Calculates scale for second item in visible area"
+  [{:keys [active-index]}]
+  (if (> active-index ITEMS_BEFORE_SCROLL)
+    MEDIUM_SCALE
+    DEFAULT_SCALE))
+
+(defn- last-item
+  "Calculates scale for last item in visible area"
+  [{:keys [active-index total-length]}]
+  (let [last-index (dec total-length)]
+    (if (> active-index (- last-index ITEMS_BEFORE_SCROLL))
+      MEDIUM_SCALE
+      SMALL_SCALE)))
+
+(defn- butlast-item
+  "Calculates scale for butlast item in visible area"
+  [{:keys [active-index total-length]}]
+  (if (> active-index (- (dec total-length) ITEMS_BEFORE_SCROLL))
+    DEFAULT_SCALE
+    MEDIUM_SCALE))
+
+(defn- get-scale
+  [index active-index total-length]
+  (let [relative-index (calc-relative-index index active-index total-length)]
+    (if (or (= index active-index)
+            (< total-length MAX_LENGTH))
+      DEFAULT_SCALE
+      (cond
+        (< relative-index 0) MICRO_SCALE
+        (= relative-index 0) (first-item {:active-index active-index})
+        (= relative-index 1) (second-item {:active-index active-index})
+        (= relative-index 4) (butlast-item {:total-length total-length
+                                            :active-index active-index})
+        (= relative-index 5) (last-item {:total-length total-length
+                                         :active-index active-index})
+        (> relative-index 5) MICRO_SCALE
+        :else
+        DEFAULT_SCALE))))
+
+(defn make-item-label
+  [index active?]
+  (keyword (str "slider-item-" index (when active? "-active"))))
+
+(defn- item
+  [{:keys [index active-index total-length active? customization-color theme blur?]
+    :or   {customization-color :blue
+           theme               :light}}]
+  (let [shared-scale (reanimated/use-shared-value (get-scale index active-index total-length))]
+    (rn/use-mount (fn []
+                    (let [new-scale-value (get-scale index active-index total-length)]
+                      (reanimated/animate shared-scale new-scale-value))))
+    [rn/view
+     {:key   index
+      :style (style/item-wrapper {:size    ITEM_SIZE
+                                  :spacing ITEM_SPACING})}
+     [reanimated/view
+      {:accessibility-label (make-item-label index active?)
+       :style
+       (reanimated/apply-animations-to-style
+        {:transform [{:scale shared-scale}]}
+        (style/item {:size                ITEM_SIZE
+                     :spacing             ITEM_SPACING
+                     :active?             active?
+                     :customization-color customization-color
+                     :theme               theme
+                     :blur?               blur?}))}]]))
+
+(defn- make-data
+  [amount]
+  (range amount))
+
+(defn- key-fn [i] i)
```

Comment:
do we really need this function? can't we use just identity function?

Reply:
In this case we can use `identity`